### PR TITLE
Upgrade to Xcode 15.1

### PR DESCRIPTION
**Pull request by BalestraPatrick**

### CODE REVIEW

File: test/apple_shell_testutils.sh [hunk 1]
```diff
@@ -523,21 +523,19 @@ function is_device_build() {
...
-  # against.
-  xcrun llvm-objdump --macho --section=__TEXT,__entitlements "$binary" | \
-      sed -e 's/^[0-9a-f][0-9a-f]*[[:space:]][[:space:]]*//' \
-          -e 'tx' -e 'd' -e ':x' | xxd -r -p
+  local output="$2"
+  local archs=($(lipo -archs "$binary"))
+  local thin_binary="tempdir/thin_binary"
+  mkdir -p tempdir
+  lipo -thin ${archs[0]} "$binary" -output "$thin_binary"
+  xcrun segedit "$thin_binary" -extract __TEXT __entitlements "$output"
```
Comment (keith) [inline test/apple_shell_testutils.sh:537]: couldn't we output this to `-` as stdout instead?

Comment (BalestraPatrick) [reply test/apple_shell_testutils.sh:537]: Happy to take suggestions though!

Comment (keith) [reply test/apple_shell_testutils.sh:537]: I was suggesting to do the lipo, and then just make the segedit part output to stdout so you don't have to make callers know about the temporary files

**Review by keith — APPROVED** [review]

File: test/apple_shell_testutils.sh (hunk 1 above)
Comment (BalestraPatrick) [reply test/apple_shell_testutils.sh:537]: I'll merge and investigate

**Review by luispadron — APPROVED** [review]

**Pull request merged**